### PR TITLE
ROX-22297: compliance results with no acs config

### PR DESCRIPTION
**Pull request by dashrews78**
## Description

Not all compliance operator scans may be configured by ACS.  We should still show their results.  This means we may have a scan config id of "" in the `/v2/compliance/scan/results` results.  Additionally using the scan configuration name to query causes problems due to our prefix matching search framework.  So this adds a specific endpoint to query by scan config name which will exact match while also allowing pagination.

Added `/v2/compliance/scan/results/{scan_config_name}` and `/v2/compliance/scan/count/results/{scan_config_name}`

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
~- [ ] Evaluated and added CHANGELOG entry if required~
~- [ ] Determined and documented upgrade steps~
~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs]~(https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

Tested locally with a cluster, executing commands such as below and verifying that I only got results for `test-1` and not for `test-1-1`.  :
```
curl --insecure https://central-stackrox.apps.ds-02-07-combative-holiday-r.ocp.infra.rox.systems:443/v2/compliance/scan/results/test-1 -u admin:**** | json_pp
curl --insecure https://central-stackrox.apps.ds-02-07-combative-holiday-r.ocp.infra.rox.systems:443/v2/compliance/scan/results/test-1?query.query=Cluster%20Id:d0221130-6704-414f-b255-e3ba250f6ebf -u admin:**** | json_pp
```

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: central/complianceoperator/v2/checkresults/service/service_impl.go [hunk 1]
```diff
@@ -131,8 +133,20 @@ func (s *serviceImpl) GetComplianceClusterScanStats(ctx context.Context, query *
...
+			if err != nil {
+				return nil, errors.Errorf("Unable to retrieve valid compliance scan configuration for results from %v", query)
+			}
+			scanConfigToIDs[result.ScanConfigName] = config.GetId()
+		}
+	}
+
 	return &v2.ListComplianceClusterScanStatsResponse{
-		ScanStats: storagetov2.ComplianceV2ClusterStats(scanResults),
+		ScanStats: storagetov2.ComplianceV2ClusterStats(scanResults, scanConfigToIDs),
```
Comment (SimonBaeumer) [inline central/complianceoperator/v2/checkresults/service/service_impl.go:149]: Is this code duplicate in multiple places to receive the scan config IDs? Could maybe be merged into a common function.

File: central/complianceoperator/v2/checkresults/service/service_impl.go [hunk 2]
```diff
@@ -205,3 +219,74 @@ func (s *serviceImpl) GetComplianceScanCheckResult(ctx context.Context, req *v2.
...
+		parsedQuery,
+	)
+
+	scanResults, err := s.complianceResultsDS.SearchComplianceCheckResults(ctx, parsedQuery)
+	if err != nil {
+		return nil, errors.Wrapf(errox.InvalidArgs, "Unable to retrieve compliance scan results for request %v", request)
+	}
+
+	// Need to look up the scan config IDs to return with the results.
+	scanConfigToIDs := make(map[string]string, len(scanResults))
```
Comment (SimonBaeumer) [inline central/complianceoperator/v2/checkresults/service/service_impl.go:252]: Duplicate code?

File: central/complianceoperator/v2/checkresults/service/service_impl_test.go [hunk 3]
```diff
@@ -412,6 +414,110 @@ func (s *ComplianceResultsServiceTestSuite) TestGetComplianceScanResult() {
...
+				returnResults := []*storage.ComplianceOperatorCheckResultV2{
+					convertUtils.GetComplianceStorageResults(s.T())[0],
+				}
+
+				s.resultDatastore.EXPECT().SearchComplianceCheckResults(gomock.Any(), expectedQ).Return(returnResults, nil).Times(1)
+				s.scanConfigDS.EXPECT().GetScanConfigurationByName(gomock.Any(), "scanConfig1").Return(getTestRec("scanConfig1"), nil).Times(1)
+			},
+		},
+		{
+			desc: "Query with no scam configuration name field",
```
Comment (SimonBaeumer) [inline central/complianceoperator/v2/checkresults/service/service_impl_test.go:493]: scam or scan? :D

**Review by SimonBaeumer — APPROVED** [review]
LGTM, only some small nits.

File: central/complianceoperator/v2/checkresults/service/service_impl.go (hunk 1 above)
Comment (SimonBaeumer) [reply central/complianceoperator/v2/checkresults/service/service_impl.go:149]: GoLand had put my comment on the wrong line. Resolving this conversation.

File: central/complianceoperator/v2/checkresults/service/service_impl.go [hunk 4]
```diff
@@ -131,8 +133,20 @@ func (s *serviceImpl) GetComplianceClusterScanStats(ctx context.Context, query *
...
+	for _, result := range scanResults {
+		if _, found := scanConfigToIDs[result.ScanConfigName]; !found {
+			config, err := s.scanConfigDS.GetScanConfigurationByName(ctx, result.ScanConfigName)
+			if err != nil {
+				return nil, errors.Errorf("Unable to retrieve valid compliance scan configuration for results from %v", query)
+			}
+			scanConfigToIDs[result.ScanConfigName] = config.GetId()
+		}
+	}
```
Comment (SimonBaeumer) [inline central/complianceoperator/v2/checkresults/service/service_impl.go:146]: Is this code duplicate in multiple places to receive the scan config IDs? Could maybe be merged into a common function.

**Review by SimonBaeumer — APPROVED** [review]

File: central/complianceoperator/v2/checkresults/service/service_impl.go (hunk 2 above)
Comment (dashrews78) [reply central/complianceoperator/v2/checkresults/service/service_impl.go:252]: yeah duplicate to code in `GetComplianceScanResults`

**Pull request merged**